app/js/trouble.js

```diff
@@ -115,9 +124,9 @@
 		});
 
 		$toggle.on('click', toggleDropdown);
-		this.open = open;
-		this.close = close;
-		this.setDropdownHeight = setDropdownHeight;
+		this.open = open();

```

Comment:
What's going on here is that using this.something means the code is actually declaring global variables - since this code doesn't appear in a method, "this" is bound to the global object. When using strict mode ("use strict";) it will complain about declaring global variables inside a function.